stable-25-1-3 + main (tsan): ydb/tests/compatibility test_compatibility.py.TestCompatibility.test_tpch1[restart_current_to_stable-25-1-3-column]

https://github.com/ydb-platform/ydb/actions/runs/16426501743

https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/16426501743/ya-main-x86-64-tsan/try_1/ya-test.html

Падает  тест под tsan 
ydb/tests/compatibility test_compatibility.py.TestCompatibility.test_tpch1[restart_current_to_stable-25-1-3-column]

```
E   Write of size 1 at 0x7214003d23d8 by thread T72:
E   #0 OnExecute /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:158:21 (ydbd-target+0x28f3bca2) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #1 NKikimr::NOlap::NReader::NCommon::TFetchingScriptCursor::Execute(std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::IDataSource> const&) /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:59:13 (ydbd-target+0x28f3bca2)
E   #2 NKikimr::NOlap::NReader::NCommon::TStepAction::DoExecuteImpl() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:31:33 (ydbd-target+0x28f3b137) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #3 NKikimr::NOlap::NReader::IDataTasksProcessor::ITask::DoExecute(std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/columnshard/engines/reader/common/conveyor_task.cpp:8:19 (ydbd-target+0x29050c1a) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #4 NKikimr::NConveyor::ITask::Execute(std::__y1::shared_ptr<NKikimr::NConveyor::TTaskSignals>, std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/conveyor/usage/abstract.cpp:16:9 (ydbd-target+0x27f6f762) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #5 NKikimr::NConveyorComposite::TWorker::ExecuteTask(std::__y1::vector<NKikimr::NConveyorComposite::TWorkerTask, std::__y1::allocator<NKikimr::NConveyorComposite::TWorkerTask>>&&) /-S/ydb/core/tx/conveyor_composite/service/worker.cpp:17:22 (ydbd-target+0x34567003) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #6 NKikimr::NConveyorComposite::TWorker::HandleMain(TAutoPtr<NActors::TEventHandle<NKikimr::NConveyorComposite::TEvInternal::TEvNewTask>, TDelete>&) /-S/ydb/core/tx/conveyor_composite/service/worker.cpp:59:5 (ydbd-target+0x34569c18) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #7 NKikimr::NConveyorComposite::TWorker::StateMain(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/core/tx/conveyor_composite/service/worker.h:39:13 (ydbd-target+0x3455be94) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #8 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:406:17 (ydbd-target+0x195326d5) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #9 NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:268:28 (ydbd-target+0x19596519) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #10 NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:458:39 (ydbd-target+0x1959c27a) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #11 NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:510:13 (ydbd-target+0x1959bbd0) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #12 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:536:9 (ydbd-target+0x1959cf72) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #13 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:386:45 (ydbd-target+0x17fbf4f2) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #14 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:245:20 (ydbd-target+0x17fc266c) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   
E   Previous read of size 1 at 0x7214003d23d8 by thread T76:
E   #0 OnExecute /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:156:14 (ydbd-target+0x28f3bc7e) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #1 NKikimr::NOlap::NReader::NCommon::TFetchingScriptCursor::Execute(std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::IDataSource> const&) /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:59:13 (ydbd-target+0x28f3bc7e)
E   #2 NKikimr::NOlap::NReader::NCommon::TStepAction::DoExecuteImpl() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:31:33 (ydbd-target+0x28f3b137) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #3 NKikimr::NOlap::NReader::IDataTasksProcessor::ITask::DoExecute(std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/columnshard/engines/reader/common/conveyor_task.cpp:8:19 (ydbd-target+0x29050c1a) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #4 NKikimr::NConveyor::ITask::Execute(std::__y1::shared_ptr<NKikimr::NConveyor::TTaskSignals>, std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/conveyor/usage/abstract.cpp:16:9 (ydbd-target+0x27f6f762) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #5 NKikimr::NConveyorComposite::TWorker::ExecuteTask(std::__y1::vector<NKikimr::NConveyorComposite::TWorkerTask, std::__y1::allocator<NKikimr::NConveyorComposite::TWorkerTask>>&&) /-S/ydb/core/tx/conveyor_composite/service/worker.cpp:17:22 (ydbd-target+0x34567003) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #6 NKikimr::NConveyorComposite::TWorker::HandleMain(TAutoPtr<NActors::TEventHandle<NKikimr::NConveyorComposite::TEvInternal::TEvNewTask>, TDelete>&) /-S/ydb/core/tx/conveyor_composite/service/worker.cpp:59:5 (ydbd-target+0x34569c18) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #7 NKikimr::NConveyorComposite::TWorker::StateMain(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/core/tx/conveyor_composite/service/worker.h:39:13 (ydbd-target+0x3455be94) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #8 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:406:17 (ydbd-target+0x195326d5) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #9 NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:268:28 (ydbd-target+0x19596519) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #10 NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:458:39 (ydbd-target+0x1959c27a) (BuildId: b561e1e91821e07124ef23d16893e0a8b5955fed)
E   #11 NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:510:13 (ydbd-target+0x1959bbd0) (BuildId: b561e1e91821e07124ef23...teardown failed:
ydb/tests/compatibility/test_compatibility.py:16: in setup
```

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
@@ -137,9 +137,8 @@ class TFetchingScript {
 private:
     YDB_READONLY_DEF(TString, BranchName);
     std::vector<std::shared_ptr<IFetchingStep>> Steps;
-    TAtomic StartInstant;
+    TAtomic StartInstant = 0;
     TAtomic FinishInstant;
-    TAtomic Started = 0;
 
 public:
     TFetchingScript(const TString& branchName, std::vector<std::shared_ptr<IFetchingStep>>&& steps)
@@ -153,10 +152,7 @@ public:
     }
 
     void OnExecute() {
-        if (!AtomicGet(Started)) {
-            AtomicSet(StartInstant, TMonotonic::Now().MicroSeconds());
-            AtomicSet(Started, 1);
-        }
+        AtomicCas(&StartInstant, 0, TMonotonic::Now().MicroSeconds());
     }
 
     TString DebugString() const;

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
@@ -152,7 +152,7 @@ public:
     }
 
     void OnExecute() {
-        AtomicCas(&StartInstant, 0, TMonotonic::Now().MicroSeconds());
+        AtomicCas(&StartInstant, TMonotonic::Now().MicroSeconds(), 0);
     }
 
     TString DebugString() const;

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
@@ -139,7 +139,7 @@ private:
     std::vector<std::shared_ptr<IFetchingStep>> Steps;
     TAtomic StartInstant;
     TAtomic FinishInstant;
-    bool Started = false;
+    TAtomic Started = 0;
 
 public:
     TFetchingScript(const TString& branchName, std::vector<std::shared_ptr<IFetchingStep>>&& steps)
@@ -153,9 +153,9 @@ public:
     }
 
     void OnExecute() {
-        if (!Started) {
+        if (!AtomicGet(Started)) {
             AtomicSet(StartInstant, TMonotonic::Now().MicroSeconds());
-            Started = true;
+            AtomicSet(Started, 1);
         }
     }
 

